platformio.ini

```diff
@@ -0,0 +1,53 @@
+; PlatformIO Project Configuration File
+;
+;   Build options: build flags, source filter
+;   Upload options: custom upload port, speed and extra flags
+;   Library options: dependencies, extra library storages
+;   Advanced options: extra scripting
+;
+; Please visit documentation for the other options and examples
+; https://docs.platformio.org/page/projectconf.html
+
+[platformio]
+src_dir=.
+
+[env]
+framework = arduino
+lib_deps = fortyseveneffects/MIDI Library@^5.0.2
+build_src_filter =
+  +<*.c>
+  +<*.cpp>
+  +<*.h>
+  +<*.ino>
+
+; see https://docs.platformio.org/en/latest/platforms/teensy.html for more details
+[env:teensy36]
+platform = teensy
+board = teensy36
+build_flags = -D USB_MIDI
+
+[env:teensy40]
+platform = teensy
+board = teensy40
+build_flags = -D USB_MIDI
+
+[env:teensy41]
+platform = teensy
+board = teensy41
+build_flags = -D USB_MIDI
+
+; see https://docs.platformio.org/en/latest/platforms/espressif32.html for more details
+[env:esp32doit-devkit-v1]
+platform = espressif32
+platform_packages = framework-arduinoespressif32 @ https://github.com/espressif/arduino-esp32.git#2.0.9
```

Comment:
Currently newer versions [are broken](https://github.com/corrados/edrumulus/blob/main/README.md?plain=1#L26).

To test newer versions, I'd suggest to change it in a branch and test then. One might want to [allow branches other than `main` for that](https://github.com/corrados/edrumulus/blob/main/.github/workflows/main.yml#L5).

Reply:
But yes, the MIDI library is also pinned to 5.0.2. Most projects want to have control over what version is used during their build process to make testing reproducible.

Reply:
Thank you thijstriemstra for your review.

3hhh, that makes totally sense what you have configured. I would actually keep your implementation. The only issue is that we have to remember to change the version to a newer version as soon as the issue in the ESP32 library is fixed since I am only using the Arduino IDE an not using your new platform IO configuration file. So, I may not notice that there is something to do in the future.